specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/virtualWan.json

```diff
@@ -2634,6 +2634,231 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/vpnGateways/{gatewayName}/vpnConnections/{connectionName}/vpnLinkConnections/{linkConnectionName}/sharedKey": {
```

Comment:
>sharedKey [](http://example.com/codeflow?start=198&length=9)

The resourceType shuld be plural. Also, is sharedKey a resource type or an action name? I can see, you have defined both GET and POST on the same sharedKey name, which is fundamentally wrong.

Reply:
We are following the API spec for Virtual Network Gateway Get Shared Key for consistency. Please refer to the below link. 

https://learn.microsoft.com/en-us/rest/api/network-gateway/virtual-network-gateway-connections/get-shared-key?view=rest-network-gateway-2023-09-01&tabs=HTTP

Example: GET https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/connections/{virtualNetworkGatewayConnectionName}/sharedkey?api-version=2023-09-01

Reply:
Unfortunately, that API doesn't follow ARM guidelines. Since you are wrapping the other API, it's not clear that consistency with a non ARM-conforming API is better than consistency with other ARM resources. To design this correctly for ARM, you could model it as a singleton resource named sharedKeys/default. You would need to support singleton GET, PUT, and a reset POST action (all on the sharedKeys/default resource), plus collection GET on the sharedKeys resource type (would always return a collection of one item). Your backend would need to slightly reshape the requests to talk to the other interface, but that should be simple.

Reply:
@mentat9 Sharing doc where we got Hybrid API review signoff [GetVpnConnectionSharedKeyAPI.docx](https://microsoft-my.sharepoint.com/:w:/p/liusheng/ESV5acTQcilMhbdQM3ZiCnIBBdbfEnZwUK4XfxhIzXgWTg?e=zutH2D)
We have a case opened for this and customers are waiting for this. Please provide a solution here

Reply:
I'm not familiar with the Hybrid API review process. Regarding ARM compliance: please consider this feedback in future API designs. Extending non-ARM compliant APIs from one RP to another is not something we want, even to maintain consistency between those two. We are trying to improve the overall Azure API surface area by raising the quality and consistency of new APIs as they are added. Not blocking on this, since you are passing through another RPs API.